test/ready/shared.dart

```diff
@@ -61,4 +62,18 @@ void sharedTests() {
     // Should be back to not ready.
     expect(watcher.ready, doesNotComplete);
   });
+
+  test('completes with error if directory does not exist', () async {
+    var watcher = createWatcher(path: 'does/not/exist');
+
+    // Subscribe to the events (else ready will never fire).
+    // ignore: unused_local_variable
+    var subscription = watcher.events.listen((event) {}, onError: (error) {});
```

Comment:
Do we also get the error on this stream still? If not you can just use `drain()`, and I think it might be best to not duplicate the error on this stream (just forces people to handle it twice).

Reply:
Yep, the error is still on the stream. That's existing behaviour and if we remove it from there, this will be a breaking change to anyone not handling errors from `ready` (which could never happen until now).

I'm not certain I need this `onError` here though, I added it while trying to debug something (resubscribable would previously `await ready` without handling the error which was failing the test) and forgot to remove it. I'll remove it if it doesn't break the test, but let me know if you think we should make the breaking change above too.

Reply:
Nope, it was also required.

Reply:
> Yep, the error is still on the stream. That's existing behaviour and if we remove it from there, this will be a breaking change to anyone not handling errors from ready (which could never happen until now).

This is still going to be breaking for those users since its a new error which might go unhandled (and crash the app etc). I have a feeling its best to also change the behavior around errors on the stream at the same time? I don't think its really _more_ breaking.